cmd/nerdctl/run_windows_test.go

```diff
@@ -65,3 +68,43 @@ func TestRunProcessIsolated(t *testing.T) {
 	containerUser := "ContainerUser"
 	base.Cmd("run", "--rm", "--isolation=process", "-u", containerUser, testutil.WindowsNano, "whoami").AssertOutContains(containerUser)
 }
+
+func TestRunHyperVContainer(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	// Hyper-V Virtual Machine Management service
+	hypervServiceName := "vmms"
+
+	m, err := mgr.Connect()
+	if err != nil {
+		t.Fatalf("unable to access Windows service control manager: %s", err)
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(hypervServiceName)
+	// hyperv service is not present, hyperv is not enabled
+	if err != nil {
+		t.Skip("HyperV is not enabled, skipping test")
+	}
+	defer s.Close()
+
+	// hyperv must not be in the name for this test, the output is parsed for it
+	containerName := "nerdctl-testwcowcontainer"
+	base.Cmd("run", "--isolation", "hyperv", "--name", containerName, testutil.WindowsNano).Out()
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+	inspectOutput := base.Cmd("container", "inspect", "--mode", "native", containerName).Out()
+
+	assert.Check(t, strings.Contains(inspectOutput, "hyperv"))
+}
+
+func TestRunProcessContainer(t *testing.T) {
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	containerName := testutil.Identifier(t)
+
+	base.Cmd("run", "--isolation", "process", "--name", containerName, testutil.WindowsNano).Out()
+	defer base.Cmd("rm", "-f", containerName).AssertOK()
+	inspectOutput := base.Cmd("container", "inspect", "--mode", "native", containerName).Out()
+
+	assert.Check(t, !strings.Contains(inspectOutput, "hyperv"))
```

Comment:
I think its safer to use assert.Assert.
https://pkg.go.dev/testing#T.FailNow (will also call defer)